File: app/src/main/java/com/studenthub/app/model/AppNotification.java
package com.studenthub.app.model;

import com.google.firebase.firestore.ServerTimestamp;
import java.util.Date;

public class AppNotification {
    private String id;
    private String title;
    private String body;
    private String text;
    private String toUid;
    private Date createdAt;

    public AppNotification() {}
    public String getId(){return id;} public void setId(String v){id=v;}
    public String getTitle(){return title;} public void setTitle(String v){title=v;}
    public String getBody(){return body;} public void setBody(String v){body=v;}
    public String getText(){return text;} public void setText(String v){text=v;}
    public String getToUid(){return toUid;} public void setToUid(String v){toUid=v;}
    @ServerTimestamp public Date getCreatedAt(){return createdAt;} public void setCreatedAt(Date v){createdAt=v;}
}

File: android/app/src/main/java/com/studenthub/app/data/model/AppNotification.kt
package com.studenthub.app.data.model

import com.google.firebase.Timestamp

data class AppNotification(
    val id: String = "",
    val toUid: String = "",
    val title: String = "",
    val body: String = "",
    val createdAtMillis: Long = 0L,
    val sentBy: String = ""
) {
    companion object {
        fun fromMap(id: String, map: Map<String, Any?>): AppNotification {
            return AppNotification(
                id = id,
                toUid = map["toUid"] as? String ?: "",
                title = map["title"] as? String ?: "",
                body = map["body"] as? String ?: "",
                createdAtMillis = (map["createdAt"] as? Timestamp)?.toDate()?.time ?: 0L,
                sentBy = map["sentBy"] as? String ?: ""
            )
        }
    }
}
